tests/plugins/pastefromword/_helpers/blob.js

```diff
@@ -0,0 +1,113 @@
+	// jscs:enable maximumLineLength
+		isSupportedEnvironment = typeof Uint8Array === 'function' && typeof Blob === 'function' &&
+			typeof URL === 'function' && typeof ArrayBuffer === 'function';
+
+	function ignoreUnsupportedEnvironment( testSuite, check ) {
+		testSuite._should = testSuite._should || {};
+		testSuite._should.ignore = testSuite._should.ignore || {};
+
+		for ( var key in testSuite ) {
+			if ( ( typeof check !== 'undefined' && !check ) || !this.isSupportedEnvironment ) {
+				testSuite._should.ignore[ key ] = true;
+			}
+		}
+	}
+
+	function dataUriToBlob( dataURI ) {
+		var base64String = atob( dataURI.split( ',' )[1] ),
+			fileType = dataURI.match( /^data:([^;]*);base64,/ )[ 1 ],
+			arrayBuffer = new Uint8Array( base64String.length );
+
+		for ( var i = 0; i < base64String.length; i++ ) {
+			arrayBuffer[ i ] = base64String.charCodeAt( i );
+		}
+
+		return new Blob( [ arrayBuffer ], { type: fileType } );
+	}
+
+	function simulatePasteBlob( editor, assertion, options ) {
+		var image = options && options.image || SAMPLE_PNG,
+			imageDataUrl = options && options.imageDataUrl || SAMPLE_PNG,
+			url = getObjectUrl( image, options && options.imageType ),
+			template = options && options.template || '<p{%CLASS%}>Hello<img style="height:200px; width:200px" src="{%URL%}" />world.</p>',
+			input = template.replace( /{%URL%}/g, url ).replace( /{%CLASS%}/g, ' class="MsoNormal"' ),
+			expected = template.replace( /{%URL%}/g, imageDataUrl ).replace( /{%CLASS%}/g, '' );
```

Comment:
The code is a bit unreadable with all default option values. Maybe we could simplify it with some merging logic?

```js
CKEDITOR.tools.object.merge( defaultOptions, options || {} );
```